src/Features/Lsif/Generator/Generator.cs

```diff
@@ -440,32 +440,21 @@ private static (DefinitionRangeTag tag, TextSpan fullRange)? CreateRangeTagAndCo
             return (new DefinitionRangeTag(syntaxToken.Text, symbolKind, fullRange), fullRangeSpan);
         }
 
-        private static async Task<(Uri uri, string? contentBase64Encoded)> GetUriAndContentAsync(
+        private static async Task<string?> GetBase64EncodedContentAsync(
             Document document, CancellationToken cancellationToken)
         {
-            Contract.ThrowIfNull(document.FilePath);
-
-            string? contentBase64Encoded = null;
-            Uri uri;
-
             if (document is SourceGeneratedDocument)
             {
                 var text = await document.GetValueTextAsync(cancellationToken);
 
                 // We always use UTF-8 encoding when writing out file contents, as that's expected by LSIF implementations.
                 // TODO: when we move to .NET Core, is there a way to reduce allocations here?
-                contentBase64Encoded = Convert.ToBase64String(Encoding.UTF8.GetBytes(text.ToString()));
-
-                // There is a triple slash here, so the "host" portion of the URI is empty, similar to
-                // how file URIs work.
-                uri = ProtocolConversions.CreateUriFromSourceGeneratedFilePath(document.FilePath);
+                return Convert.ToBase64String(Encoding.UTF8.GetBytes(text.ToString()));
```

Comment:
why are we doing base64?

Reply:
The LSIF spec allows for embedding document contents, and this is how it's specced.